packages/wallets/src/features/cashier/modules/Transactions/Transactions.tsx

```diff
@@ -38,10 +34,14 @@ const Transactions = () => {
     const { isMobile } = useDevice();
 
     const { location } = useHistory();
-    const state: THistoryState = location.state;
+    const initialShowPending = Boolean(
+        location.pathname === '/wallets/cashier/transactions' ? location.state?.showPending : false
+    );
+    const initialTransactionType =
+        (location.pathname === '/wallets/cashier/transactions' ? location.state?.transactionType : undefined) ?? 'all';
```

Comment:
These are type checks that narrow the type of `location.state` for TS to know that it's related to `/wallets/cashier/transactions` route and hence contains the `showPending` and `transactionType` props